### ISSUE TITLE
Reg Stats project structure

### ISSUE THREAD

**Issue by mfebrizio** [regulatorystudies/Reg-Stats#6]
First off, I changed the name of the repo to Reg-Stats, so it's more in line with the page on our [website](https://regulatorystudies.columbian.gwu.edu/reg-stats).

Currently, this repository just has R code for creating charts. Should we keep it as such, or would it also be worth integrating other code that is used for the data collection too? For example, I have code I use for retrieving and formatting the Federal Register rules data.

Seems like there are two main options:

1. If we keep this repo just for charts, then we should create separate repos for each data collection/processing regimen (if code is involved).

2. If we opt to integrate them all into one repo, we should restructure this one. Something like renaming the "code" folder to be called "charts", creating new folders for data sources like "federal_register", "reginfo", etc. Stuff like environment files, LICENSE, README, etc. should stay at the project root though.

What do each of you think? I want to start using this space to handle topics related to Reg-Stats, so it felt appropriate to create an "issue" in this case.

**Comment by zhoudanxie** [comment]
I prefer managing everything in one repo. Does it make sense to have a similar structure to what we have in Box? E.g., under "es_rules_published" folder, we'll have data, code for creating charts, chart PDF, etc.

**Comment by mfebrizio** [comment]
That makes sense. I think we could structure it that way, as long as stuff like renv/, .Rproj, LICENSE, etc. stay in the root.

- root/
  - chart_1/
    - code/
    - data/
    - output/
  - chart_2/
    - code/
    - data/
    - output/

Or maybe no subfolders within each charts folder, and the files are contained there directly.

**Comment by zhoudanxie** [comment]
That sounds good to me!

*Zhoudan (Zoey) Xie*
Senior Policy Analyst
Regulatory Studies Center <https://regulatorystudies.columbian.gwu.edu/>
The George Washington University
Website: zhoudanxie.github.io

**Comment by zhoudanxie** [comment]
Oh I see. That makes sense. How about grouping the data and code for
collecting the data into a "data" folder? So it will be:

   - root
      - charts/
         - code
         - output
      - data/
         - es_rules_published/
            - data
            - code for collecting the data
         - fr_rules/
            - data
            - code for collecting the data
         - etc...

Best,
Zoey

*Zhoudan (Zoey) Xie*
Senior Policy Analyst
Regulatory Studies Center <https://regulatorystudies.columbian.gwu.edu/>
The George Washington University
Website: zhoudanxie.github.io

**Comment by mfebrizio** [comment]
Ah yes, I think that structure would work!

Anyone else have thoughts on this?

**Comment by henhirsch** [comment]
I think that sounds good.

We also might want to include Word docs containing instructions on how to update the CSV and Rmd files/generate the graphs.

**Comment by mfebrizio** [comment]
Yes, a couple ideas there:

- a documentation/ folder within the root
- include the update instructions in the README within the root (my preferred option)
- another instruction document / README within the charts folder

These can be paired with instructions within the ipynb or Rmd files.

**Comment by zhoudanxie** [comment]
I like using README for instructions. We may need multiple files for the
instructions for updating data, since the updating process is different for
each data series. We can have one README for updating charts. How about
this:

   - root
      - charts/
         - code
         - output
         - README (instructions for updating charts)
      - data/
         - es_rules_published/
            - data
            - code for collecting the data (if applicable)
            - README (instructions for updating data)
         - fr_rules/
            - data
            - code for collecting the data (if applicable)
            - README (instructions for updating data)
         - etc...
      - README (about the repo)


*Zhoudan (Zoey) Xie*
Senior Policy Analyst
Regulatory Studies Center <https://regulatorystudies.columbian.gwu.edu/>
The George Washington University
Website: zhoudanxie.github.io

**Comment by henhirsch** [comment]
One potential problem: in some cases the code needs to be updated as well. For example, adding presidential names for the Cumulative ES Rules by Admin code/graph. However, this should be a rare occurrence so we may just be able to add specific comments in the relevant code(instead of separate README files for each ipynb or Rmd file).

**Comment by mfebrizio** [comment]
Yeah, I prefer README as well (or rather markdown; the file could be called anything really). I like the structure @zhoudanxie suggested.

If I understand your point, @henhirsch, you're saying that it wouldn't be ideal to include details on updating the code within the relevant README files, even if it's a rare occurrence. I agree with that. I wonder if we could include a step in each set of updating instructions along the lines of: "Make revisions to code, as needed, to account for changes (e.g., a new president takes office)." Then more specific comments/instructions could be in the Rmd/ipynb file on what might need to be adjusted.

**Comment by henhirsch** [comment]
Yes, that’s what I meant. We need to provide custom update instructions for many of the scripts, but it’s too cumbersome to have separate README files for each of them.

**Comment by zhoudanxie** [comment]
That all sounds good to me. To implement these changes, I think it would
make sense to have the person who created the code write the chart update
instructions -- I guess it's mostly Henry and Yaxin? Then Mark, Sarah and I
can work on restructuring the repo and moving the data update instructions
from Box to Github. We can discuss in detail in our Friday meeting.

*Zhoudan (Zoey) Xie*
Senior Policy Analyst
Regulatory Studies Center <https://regulatorystudies.columbian.gwu.edu/>
The George Washington University
Website: zhoudanxie.github.io
